### PR TITLE
Added the automation test case with validations for PG for co-dependent schemas

### PR DESCRIPTION
**Pull request by priyanshi-yb**
1. Added the tests for co-dependent schemas
2. Added a case in `pg-partitions` for Range Partition with multiple columns and also update the validations with more validations check for `pg-partitions`.

### CODE REVIEW

File: migtests/tests/pg-tests/pg-codependent-schemas/validate [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	'v1': 2,
+}
+
+def check_validations_per_schema(tgt,schema_name):
+
+	EXPECTED_ROW_COUNT = EXPECTED_ROW_COUNT_PUBLIC if schema_name == 'public' else EXPECTED_ROW_COUNT_SCHEMA2
+
+	table_list = tgt.get_table_names(schema_name)
+	print("table_list:", table_list)
+	assert (len(table_list) == 7 and schema_name == 'public') or (len(table_list) == 4 and schema_name == 'schema2')
```
Comment (amit-yb) [inline migtests/tests/pg-tests/pg-codependent-schemas/validate:36]: I see many such conditions. This defeats the purpose of creating a general function. Arrange the expected values in a dictionary keyed by the schema name:
```
EXPECTED = {
  "public": {
    "table_count": 7,
    "mview_count": 1,
  },
  "schema2": {
    "table_count": 4,
    "mview_count": 0,
  }
}
```
Then you can easily write a condition as:
```
assert len(table_list) == EXPECTED[schema_name]["table_count"]
assert len(mview_list) == EXPECTED[schema_name]["mview_count"]
```

File: migtests/tests/pg-tests/pg-codependent-schemas/validate [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	assert len(view_list) == 1
+
+	mview_list = tgt.get_objects_of_type("MVIEW", schema_name)
+	print("mview_list:", mview_list)
+	assert (len(mview_list) == 1 and schema_name == 'public') or (len(mview_list) == 0 and schema_name == 'schema2')
+
+	for table_name, row_count in EXPECTED_ROW_COUNT.items():
+		cnt = tgt.get_row_count(table_name,schema_name)
+		print(f"table_name: {schema_name}.{table_name}, row_count: {cnt}")
+		assert row_count == cnt
```
Comment (amit-yb) [inline migtests/tests/pg-tests/pg-codependent-schemas/validate:49]: You can compare two full dictionaries to have exactly the same set of key-value pairs simply using `==` operator. No need to compare each element individually.

You can rename `cnt` with a more pronounciable `count`.

Space after comma, always: you can go through the PEP8 once (https://peps.python.org/pep-0008/). You can also consider running pylint on your scripts to find such errors.

File: migtests/tests/pg-tests/pg-codependent-schemas/validate [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	
+	fetched_triggers = tgt.fetch_all_triggers(schema_name)
```
Comment (amit-yb) [inline migtests/tests/pg-tests/pg-codependent-schemas/validate:51]: How about naming the variable as just `triggers`?

File: migtests/tests/pg-tests/pg-codependent-schemas/validate [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	print(f"Checking Foreign key test on schema {schema_name} - {chk_err_returned}")
+	assert chk_err_returned == True
+
+
+def migration_completed_checks(tgt):
+
+	check_validations_per_schema(tgt, "public")
+	check_validations_per_schema(tgt, "schema2")
+	
+	cnt_partitions = tgt.execute_query("SELECT count(DISTINCT tableoid::regclass) FROM list_part;")
```
Comment (amit-yb) [inline migtests/tests/pg-tests/pg-codependent-schemas/validate:71]: Do not abbreviate `count` as `cnt`. Always try to have names that are easier to pronounce.

Let's avoid running ad-hoc queries using the `tgt`. Define a method in `tgt` to get a list of names of partitions of a given <schema, table>. And check the `len(partition_list)`.

File: migtests/tests/pg-tests/pg-codependent-schemas/validate [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	'v1':2,
+	'mv1':2,
+}
+EXPECTED_ROW_COUNT_SCHEMA2 = {
+	'audit': 4,
+	'foreign_test': 3,
+	'list_others': 2, 
+	'primary_test': 2,
+	'v1': 2,
+}
```
[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply migtests/tests/pg-tests/pg-codependent-schemas/validate:28]: Yeah, thanks!

**Review by amit-yb — APPROVED** [review]
LGTM. With couple of minor nits.

**Pull request merged**